### PR TITLE
Minimise mumber of calls to folder create

### PR DESCRIPTION
**Pull request by nakulkar-msft**

### CODE REVIEW

**Review by pranavmalik-msft — DISMISSED** [review]
LGTM

**Review by tasherif-msft — DISMISSED** [review]

File: e2etest/zt_basic_copy_sync_remove_test.go [hunk 1]
```diff
@@ -192,19 +192,15 @@ func TestBasic_CopyDownloadDir(t *testing.T) {
 }
 
 func TestBasic_CopyS2SDir(t *testing.T) {
-	RunScenarios(t, eOperation.CopyAndSync(), eTestFromTo.AllS2S(), eValidate.AutoPlusContent(), anonymousAuthOnly, anonymousAuthOnly, params{
+	RunScenarios(t, eOperation.Copy(), eTestFromTo.Other(common.EFromTo.FileBlob()), eValidate.AutoPlusContent(), anonymousAuthOnly, anonymousAuthOnly, params{
```
Comment (gapra-msft) [inline e2etest/zt_basic_copy_sync_remove_test.go:195]: why are we removing some test cases? are they duplicates?

File: ste/sender-azureFile.go [hunk 2]
```diff
@@ -531,39 +531,28 @@ func (d AzureFileParentDirCreator) CreateParentDirToRoot(ctx context.Context, fi
 }
 
 func (d AzureFileParentDirCreator) CreateDirToRoot(ctx context.Context, shareClient *share.Client, directoryClient *directory.Client, t FolderCreationTracker) error {
```
Comment (gapra-msft) [inline ste/sender-azureFile.go:533]: any way we can make useful utilities like this more testable?

Comment (nakulkar-msft) [reply ste/sender-azureFile.go:533]: These are tested via e2e framework, we use them to create resources

**Review by gapra-msft — APPROVED** [review]

**Review by vibhansa-msft — APPROVED** [review]

File: ste/folderCreationTracker.go [hunk 3]
```diff
@@ -85,6 +85,11 @@ func (f *jpptFolderTracker) CreateFolder(folder string, doCreation func() error)
 		return nil // Never persist to dev-null
 	}
 
+	if idx, ok := f.contents[folder]; ok &&
```
Comment (gapra-msft) [inline ste/folderCreationTracker.go:88]: is there any way we could add some way to test that the minimal number of and expected calls are made to folder create? Perhaps through a mock folder creation tracker in test?

**Review by vibhansa-msft — DISMISSED** [review]

**Review by vibhansa-msft — APPROVED** [review]

**Review by gapra-msft — APPROVED** [review]

**Review by tasherif-msft — APPROVED** [review]

**Pull request merged**